apps/meteor/app/livechat/imports/server/rest/departments.ts

```diff
@@ -28,8 +30,15 @@ API.v1.addRoute(
 	},
 	{
 		async get() {
-			const { offset, count } = this.getPaginationItems();
-			const { sort } = this.parseJsonQuery();
+			const { offset, count } = await getPaginationItems(this.queryParams);
+			const { sort } = await parseJsonQuery(
+				this.request.route,
+				this.userId,
+				this.queryParams,
+				this.logger,
+				this.queryFields,
+				this.queryOperations,
+			);
```

Comment:
I didn't like that we'd to pass these many props to this method which is getting used a lot of the times across the codebase. Considering all these props are already part of the generic API class, my suggestion would be to move this function (i.e. `parseJsonQuery` ), inside this API class. That way, on the endpoints, we'd only require to call `const { sort } = await this.parsedJsonQuery()`. 

Let me know you thoughts on this Marcos :)

Reply:
It makes sense, but the big problem is that it is a huge function, not sure if it is worth moving all the code inside the class (which is also a very large file), but I can do it as a wrapper maybe. Keep calling the external function, but having that class method as you suggest, wdyt?